torch/csrc/utils/python_strings.h

```diff
@@ -54,3 +55,49 @@ inline bool THPUtils_isInterned(PyObject* obj) {
 inline void THPUtils_internStringInPlace(PyObject** obj) {
   PyUnicode_InternInPlace(obj);
 }
+
+/*
+ * Reference: https://github.com/numpy/numpy/blob/f4c497c768e0646df740b647782df463825bfd27/numpy/core/src/common/get_attr_string.h#L42
+ *
+ * Stripped down version of PyObject_GetAttrString,
+ * avoids lookups for None, tuple, and List objects,
+ * and doesn't create a PyErr since this code ignores it.
+ *
+ * This can be much faster then PyObject_GetAttrString where
+ * exceptions are not used by caller.
+ *
+ * 'obj' is the object to search for attribute.
+ *
+ * 'name' is the attribute to search for.
+ *
+ * Returns a py::object wrapping the return value. If the attribute lookup failed
+ * the value will be NULL.
+ *
+ */
+
+static py::object PyObject_FastGetAttrString(PyObject *obj, char *name)
+{
+    PyTypeObject *tp = Py_TYPE(obj);
```

Comment:
We're possibly leaking `tp` here.

Reply:
How so? (This is the same as what numpy does, FWIW.)

Reply:
Never mind, I was thinking of [`PyObject_Type`](https://docs.python.org/3/c-api/object.html?highlight=pyobject_type#c.PyObject_Type) instead of [`Py_TYPE`](https://docs.python.org/3/c-api/structures.html?highlight=py_type#c.Py_TYPE).